src/hotspot/share/gc/shared/partialArraySplitter.hpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_GC_SHARED_PARTIALARRAYSPLITTER_HPP
+#define SHARE_GC_SHARED_PARTIALARRAYSPLITTER_HPP
+
+#include "gc/shared/partialArrayState.hpp"
+#include "gc/shared/partialArrayTaskStats.hpp"
+#include "gc/shared/partialArrayTaskStepper.hpp"
+#include "oops/oop.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/macros.hpp"
+
+class outputStream;
+
+// Helper class for splitting the processing of a large objArray into multiple
+// tasks, to permit multiple threads to work on different pieces of the array
+// in parallel.
+class PartialArraySplitter {
+  PartialArrayStateAllocator _allocator;
+  PartialArrayTaskStepper _stepper;
+  TASKQUEUE_STATS_ONLY(PartialArrayTaskStats _stats;)
+
+public:
+  explicit PartialArraySplitter(PartialArrayStateManager* manager,
```

Comment:
Why `explicit` for a method that has two args.

Reply:
Forgot to remove when 2nd argument added.  Originally that number from the manager, but
a potentially long-lived and reused manager with dynamic selection of worker threads made
that wrong.